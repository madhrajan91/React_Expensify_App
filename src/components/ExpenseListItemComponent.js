
import React from 'react';


import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';


const ExpenseListItemComponent = (props) => (
    <div>
        <p><Link to={`/edit/${props.expense.id}`}>{props.expense.description}</Link> </p>
        <p>
            {numeral(props.expense.amount/100).format('$0,0.00')} 
            {moment(props.expense.createdAt).format('MMMM Do, YYYY')}
        </p>
        
        </div>
);

export default ExpenseListItemComponent;