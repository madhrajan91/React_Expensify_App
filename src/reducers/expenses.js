// Expenses REducer
const expensesReducerDefaultState = []
const expensesReducer = (state= expensesReducerDefaultState, action) => {
    switch(action.type) {
        case 'ADD_EXPENSE':
            //return state.concat(action.expense); // don't use push since push adds to the original while concat returns a new object (Could   USE SPREAD instead)
            return [... state, action.expense]; // SPREAD
        case 'REMOVE_EXPENSE':
            const expenses = state.filter((expenseToRemove) => {
                return (expenseToRemove.id != action.id);
            })
            return expenses;
        case 'EDIT_EXPENSE':
            return state.map((expense)=> {
                if (expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }
                }
                else {
                    return expense;
                }
            })
        default:
            return state;
    }
}

export default expensesReducer;