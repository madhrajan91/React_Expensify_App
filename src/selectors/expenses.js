const getVisibleExpenses = (expenses, {text, sortBy, startDate, endDate}) => {
    const filteredExpenses = expenses.filter((expense) => {
       console.log(text)
       const startDateMatch = typeof startDate !== 'number' || expense.createdAt >= startDate;
       const endDateMatch = typeof endDate !== 'number' || expense.createdAt <= endDate;
       const textMatch = text === '' || expense.description.toLowerCase().includes(text.toLowerCase()) ;

       return startDateMatch && endDateMatch && textMatch;
   })
       console.log(sortBy)
       return filteredExpenses.sort((a,b) => {
           if (sortBy === 'date') {
               return a.createdAt > b.createdAt?1:-1;
           }
           else {
               return a.amount > b.amount?1:-1;
           }
       })

   //return expenses;
}

export default getVisibleExpenses;