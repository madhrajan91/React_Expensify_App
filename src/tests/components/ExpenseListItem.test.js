import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItemComponent from '../../components/ExpenseListItemComponent';
import expenses from '../fixtures/expenses';

test ('should render expenselist from expenses', () => {
    const wrapper = shallow(<ExpenseListItemComponent expense={expenses[0]}/>);
    expect(wrapper).toMatchSnapshot()
})