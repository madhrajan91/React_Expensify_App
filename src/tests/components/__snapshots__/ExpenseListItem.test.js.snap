// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenselist from expenses 1`] = `
<div>
  <p>
    <Link
      replace={false}
      to="/edit/1"
    >
      Water bill
    </Link>
     
    100
     
    1000
  </p>
</div>
`;
